dmapiclient/base.py

```diff
@@ -182,6 +183,21 @@ def _add_request_id_header(self, headers):
         headers[header] = request.request_id
         return headers
 
+    def _add_zipkin_tracing_headers(self, headers):
+        if not has_request_context():
+            return headers
+
+        span_id = request.headers.get('X-B3-SpanId', None)
+        trace_id = request.headers.get('X-B3-TraceId', None)
+
+        if span_id:
+            headers['X-B3-SpanId'] = span_id
+
+        if trace_id:
```

Comment:
Splitting hairs but seeing as CF router won't use these headers properly unless both are present in the request, this check could be `if span_id and trace_id:`.